ucsmsdk/utils/comparesyncmo.py

```diff
@@ -0,0 +1,551 @@
+        if not hasattr(to_mo, prop):
+            log.debug("Property '%s' of '%s' does not exist in diff obj." % (
+                prop, from_mo.dn))
+            continue
+
+        if getattr(from_mo, prop) != getattr(to_mo, prop):
+            diff.append(prop)
+
+    # comparing unknown properties of ref mo
+    from_xtra_props = from_mo._ManagedObject__xtra_props
+    to_xtra_props = to_mo._ManagedObject__xtra_props
+    if not to_xtra_props:
+        return
+    for prop in from_xtra_props:
+        if prop not in to_xtra_props:
+            continue
+
+        if from_xtra_props[prop].value != to_xtra_props[prop].value:
+            if version_filter:
+                UcsWarning("Ignoring xtra property '%s' of '%s'" % (
+                    prop, from_mo.dn))
+            else:
+                diff.append(prop)
+
+
+def _compare_unknown_mo(from_mo, to_mo, diff, version_filter):
+    """
+    Internal function to compare if any or both of the  ref and diff obj is
+    unknown mo.
+    """
+
+    # both unknown mo
+    if version_filter:
+        return
+    for prop in from_mo.properties:
+        if prop not in to_mo.properties:
+            continue
+        if from_mo.properties[prop] != to_mo.properties[prop]:
+            diff.append(prop)
+
+
+def _compare(from_mo, to_mo, diff, include_operational=False,
+             version_filter=True):
+    """
+    Internal method to support compare reference and difference object.
+    """
+
+    # compare mo of different types
+    if from_mo.get_class_id() != to_mo.get_class_id():
+        return _CompareStatus.TYPES_DIFFERENT
+
+    # compare for unknown class_id
+    if isinstance(from_mo, GenericMo) or isinstance(to_mo, GenericMo):
+        _compare_unknown_mo(from_mo, to_mo, diff, version_filter)
+
+    # compare for known class_id
+    else:
+        _compare_known_mo(from_mo, to_mo, diff, include_operational,
+                          version_filter)
+
+    if len(diff) > 0:
+        return _CompareStatus.PROPS_DIFFERENT
+
+    return _CompareStatus.EQUAL
+
+
+def _compare_common_mo(ref_dict, diff_dict, include_operational=False,
+                       version_filter=True, include_equal=False,
+                       exclude_different=False):
+
+    diff_output = []
+    common_dns = set(ref_dict) & set(diff_dict)
+    for dn in common_dns:
+        ref_mo = ref_dict[dn]
+        diff_mo = diff_dict[dn]
+        diff_props = []
+
+        # compare both mo for property and type
+        diff_status = _compare(ref_mo, diff_mo, diff_props,
+                               include_operational, version_filter)
+
+        if diff_status == _CompareStatus.EQUAL and include_equal:
+            mo_diff = _MoDiff(ref_mo, _MoDiff.EQUAL)
+            diff_output.append(mo_diff)
+            continue
+
+        if exclude_different:
+            continue
+
+        if diff_status == _CompareStatus.TYPES_DIFFERENT:
+            mo_diff = _MoDiff(ref_mo, _MoDiff.REMOVE)
```

Comment:
Yes, You are right, but only diff_mo is changed during translate. so making clone of diff_mo should suffice. No need to make clone of ref_mo.